Test/Makefile

```diff
@@ -0,0 +1,427 @@
+#Spin integration tests, based on the README_tests.txt and the provided examples
+
+BASE_DIR=../..
+SPIN?=${BASE_DIR}/Src/spin
+EXAMPLE_MODEL_DIR=${BASE_DIR}/Examples
+USER_TYPES_MODEL_DIR=${BASE_DIR}/Test/UserTypes
+TEST_DIR=Out
+
+CC?=gcc
+PAN=pan
+CREATE_TEST_DIR=mkdir -p ${TEST_DIR}
+DELETE_TEST_DIR=rm -r ${TEST_DIR}
+
+EXAMPLE_TESTS= \
+	test_hello \
+	test_loops \
+	test_loops_cycles \
+	test_loops_cycles_playback \
+	test_loops_cycles_avoid_progress \
+	test_partial_order_reduction_enabled \
+	test_partial_order_reduction_disabled \
+	test_rng \
+	test_ltl_generation
+
+TYPEDEF_TESTS= \
+	test_nested_typedef \
+	test_typedef_for_channel \
+	test_typedef_for_global_data \
+	test_typedef_for_local_data
+
+UNION_TESTS= \
+	test_nested_union \
+	test_union_for_channel \
+	test_union_for_global_data \
+	test_union_for_local_data
+
+COMPLEX_TESTS = \
+	test_message
+
+BENCHMARK = \
+	benchmark_large_message
+
+.SILENT : ${EXAMPLE_TESTS} \
+	${TYPEDEF_TESTS} \
+	${UNION_TESTS} \
+	${COMPLEX_TESTS} \
+	${BENCHMARK} \
+	test_rng_files \
+	check_examples \
+	check_unions \
+	check_typedefs \
+	check_complex \
+	benchmark
+
+.PHONY : ${EXAMPLE_TESTS} \
+	${TYPEDEF_TESTS} \
+	${UNION_TESTS} \
+	${COMPLEX_TESTS} \
+	${BENCHMARK} \
+	test_rng_files \
+	check_examples \
+	check_unions \
+	check_typedefs \
+	check_complex \
+	benchmark \
+	check
+
+check: ${EXAMPLE_TESTS} ${TYPEDEF_TESTS} ${UNION_TESTS} ${COMPLEX_TESTS}
+
+check_examples: ${EXAMPLE_TESTS}
+
+check_unions: ${UNION_TESTS}
+
+check_typedefs: ${TYPEDEF_TESTS}
+
+check_complex: ${COMPLEX_TESTS}
+
+benchmark: ${BENCHMARK}
+
+############################EXAMPLE TESTS######################################
+
+#Test 1: check that you can run a simulation
+test_hello:
+	echo "Test 1: check that you can run a simulation"
+	${CREATE_TEST_DIR}
+	cd ${TEST_DIR} && \
+	${SPIN} -run ${EXAMPLE_MODEL_DIR}/hello.pml > hello.output
+	grep -q "State-vector 12 byte, depth reached 2, errors: 0" ${TEST_DIR}/hello.output
+	grep -q "3 states, stored" ${TEST_DIR}/hello.output
+	${DELETE_TEST_DIR}
+
+#Test 2: a basic reachability check (safety)
+test_loops:
+	echo "Test 2: a basic reachability check (safety)"
+	${CREATE_TEST_DIR}
+	cd ${TEST_DIR} && \
+	${SPIN} -a ${EXAMPLE_MODEL_DIR}/loops.pml && \
+	${CC} -DNOREDUCE -o ${PAN} pan.c && \
+	./${PAN} > loops.output
+	grep -q "State-vector 12 byte, depth reached 11, errors: 0" ${TEST_DIR}/loops.output
+	grep -q "15 states, stored" ${TEST_DIR}/loops.output
+	grep -q "4 states, matched" ${TEST_DIR}/loops.output
+	grep -q "19 transitions (= stored+matched)" ${TEST_DIR}/loops.output
+	grep -q "0 atomic steps" ${TEST_DIR}/loops.output
+	${DELETE_TEST_DIR}
+
+#Test 3: cycle detection check (liveness)
+test_loops_cycles:
+	echo "Test 3: cycle detection check (liveness)"
+	${CREATE_TEST_DIR}
+	cd ${TEST_DIR} && \
+	${SPIN} -a ${EXAMPLE_MODEL_DIR}/loops.pml && \
+	${CC} -DNOREDUCE -o ${PAN} pan.c && \
+	./${PAN} -a > loops.output
```

Comment:
PAN=./pan (at the beginning of the file and)
${PAN)

Reply:
No, becase pan is also used for other things.